src/network/clientpackethandler.cpp

```diff
@@ -897,23 +897,43 @@ void Client::handleCommand_SpawnParticle(NetworkPacket* pkt)
 	std::string texture     = deSerializeLongString(is);
 	bool vertical           = false;
 	bool collision_removal  = false;
+	u32 material_type_param = 8464;
+	u16 vertical_frame_num = 1;
+	u16 horizontal_frame_num = 1;
+	u16 first_frame = 0;
+	float frame_length = -1;
+	bool loop_animation = true;
+	video::SColor glow(255,0,0,0);
 	try {
 		vertical = readU8(is);
 		collision_removal = readU8(is);
+		material_type_param = readU32(is);
+		vertical_frame_num = readU16(is);
+		horizontal_frame_num = readU16(is);
+		first_frame = readU16(is);
+		frame_length = readF1000(is);
+		loop_animation = readU8(is);
+		glow = readARGB8(is);
 	} catch (...) {}
 
 	ClientEvent event;
-	event.type                              = CE_SPAWN_PARTICLE;
-	event.spawn_particle.pos                = new v3f (pos);
-	event.spawn_particle.vel                = new v3f (vel);
-	event.spawn_particle.acc                = new v3f (acc);
-	event.spawn_particle.expirationtime     = expirationtime;
-	event.spawn_particle.size               = size;
-	event.spawn_particle.collisiondetection = collisiondetection;
-	event.spawn_particle.collision_removal  = collision_removal;
-	event.spawn_particle.vertical           = vertical;
-	event.spawn_particle.texture            = new std::string(texture);
-
+	event.type					= CE_SPAWN_PARTICLE;
```

Comment:
After the first letter, always indent with spaces, otherwise it will look ugly with different tab sizes. (Same for your Lua scripts)